src/main/kotlin/com/bridgecrew/ui/buttons/FixButton.kt

```diff
@@ -1,30 +1,62 @@
 package com.bridgecrew.ui.buttons
 
 import com.bridgecrew.results.BaseCheckovResult
-import com.bridgecrew.utils.getOffsetReplaceByLines
-import com.bridgecrew.utils.updateFile
+import com.bridgecrew.utils.navigateToFile
+import com.intellij.ide.DataManager
 import com.intellij.openapi.application.ApplicationManager
-import com.intellij.openapi.project.ProjectManager
+import com.intellij.openapi.command.WriteCommandAction
+import com.intellij.openapi.diagnostic.logger
+import com.intellij.openapi.editor.Document
+import com.intellij.openapi.fileEditor.FileDocumentManager
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.vfs.LocalFileSystem
+import com.intellij.openapi.vfs.VirtualFile
 import java.awt.event.ActionEvent
 import java.awt.event.ActionListener
 import javax.swing.JButton
-import javax.swing.JOptionPane
+
 
 class FixButton(val result: BaseCheckovResult) : JButton(), ActionListener {
 
+    private val LOG = logger<FixButton>()
+
     init {
         text = "Fix"
         addActionListener(this)
     }
 
     override fun actionPerformed(e: ActionEvent?) {
-        JOptionPane.showMessageDialog(null, "Fix clicked")
-        val project = ProjectManager.getInstance().defaultProject
+        this.isEnabled = false
         ApplicationManager.getApplication().invokeLater {
-            if(result.fixDefinition != null){
-                val (start, end) = getOffsetReplaceByLines(result.fileLineRange, project)
-                updateFile(result.fixDefinition, project, start, end)
+            applyFixDefinition()
+        }
+    }
+
+    private fun applyFixDefinition() {
+        try {
+            val startLine: Int = result.fileLineRange.getOrElse(0) { 1 } - 1
+            val endLine: Int = result.fileLineRange.getOrElse(1) { 1 } - 1
+
+            val virtualFile: VirtualFile = LocalFileSystem.getInstance().findFileByPath(result.absoluteFilePath)
+                    ?: return
+            val document: Document? = FileDocumentManager.getInstance().getDocument(virtualFile)
+
+            val startOffset = document!!.getLineStartOffset(startLine)
+            val endOffset = document.getLineEndOffset(endLine)
```

Comment:
we're doing this code of getting the Editor in several places now, maybe should be moved to fileUtils